src/user/user.service.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  ConflictException,
+  Injectable,
+  NotFoundException,
+  Post,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import * as bcrypt from 'bcrypt';
+import { JwtService } from '@nestjs/jwt';
+
+import { IUser } from './user.model';
+
+@Injectable()
+export class UserService {
+  constructor(
+    @InjectModel('User') private readonly userModel: Model<IUser>,
+    private jwtService: JwtService,
+  ) {}
+
+  createUser = async (user: IUser) => {
+    const saltOrRounds = 10;

```

Comment:
No error handling in whole file, and I can't see any app level error handling. Use try/catch and also introduce app level error handler